kerlescan/view_helpers.py

```diff
@@ -68,20 +68,26 @@ def ensure_has_permission(**kwargs):
 
     auth_key = get_key_from_headers(request.headers)
     if auth_key:
-        perms = get_perms(
-            kwargs["application"],
-            auth_key,
-            kwargs["logger"],
-            kwargs["request_metric"],
-            kwargs["exception_metric"],
-        )
-        for p in perms:
-            if p in kwargs["permissions"]:
-                return  # allow
-        raise HTTPError(
-            HTTPStatus.FORBIDDEN,
-            message="user does not have access to %s" % kwargs["permissions"],
-        )
+        try:
+            perms = get_perms(
+                kwargs["application"],
+                auth_key,
+                kwargs["logger"],
+                kwargs["request_metric"],
+                kwargs["exception_metric"],
+            )
+            for p in perms:
+                if p in kwargs["permissions"]:
+                    return  # allow
+            raise HTTPError(
+                HTTPStatus.FORBIDDEN,
+                message="user does not have access to %s" % kwargs["permissions"],
+            )
+        except RBACDenied:
```

Comment:
Is this try/except here necessary when you already raise the HTTP error above if you don't return out of the loop allowing access?  Are you just checking for some other form of failure when calling get_perms?

Reply:
The get_perms method can raise RBACDenied exception.  Without the try-except around the get_perms() call, the RBACDenied exception would just bubble up and likely result in a 500 error.